Lib/test/test_logging.py

```diff
@@ -4406,6 +4406,29 @@ def test_basic(self):
         self.assertTrue(found, msg=msg)
 
 
+class CachingTest(BaseTest):
```

Comment:
You could add this to an existing test case class rather than adding a separate class just for this one method.